### DISCUSSION TITLE
#[MapEntity] to fetch MongoDB Documents?

### DISCUSSION THREAD

**Discussion by icvij** [symfony/symfony#48812] [Q&A] [answered]
Hey everyone, I'm upgrading my project to Symfony 6.2 and removing `Sensio/Framework-Extra-Bundle` but there is an issue where I can't specify my custom `DocumentFinder` in `#[MapEntity]` I have also posted StackOverflow question [here](https://stackoverflow.com/questions/74942113/symfony-6-2-mapentity-to-use-custom-finder).

With `Sensio/Framework-Extra-Bundle #[ParamConverter]` that would look something like this: 
```
#[Route('/{slug}', methods: ['GET'])]
#[ParamConverter('slug', options: ['finder' => 'CustomFinder', 'attributes' => ['id']])]
```

Now I've been trying out to make this work with `#[MapEntity]` but no luck:
```
#[MapEntity(mapping: ['finder' => 'CustomFinder', 'slug' => 'id'])] MyDocument $document
```
This results in error:
```
Uncaught Error: App\Controller\MyController::get(): Argument #1 ($document) must be of type App\Document\MyDocument, string given, called in /app/vendor/symfony/http-kernel/HttpKernel.php on line 163
```

**Comment by lyrixx** [comment, marked as answer]
I think MongoDB is not supported by this new attribute

**Reply by icvij** [reply]
Thanks for the answer, not sure if i should make issue requesting a feature? I think this is a must to have feature?

**Reply by derrabus** [reply]
You can discuss and contribute the feature here: https://github.com/doctrine/DoctrineMongoDBBundle/

**Comment by icvij** [comment]
Bump

**Comment by dpi** [comment]
Can be found at doctrine/DoctrineMongoDBBundle#754
